catalog/resourcemanagement/resourcemanagement-usage-ui/src/main/java/org/codice/ddf/resourcemanagement/usage/service/DataUsage.java

```diff
@@ -0,0 +1,260 @@
+import javax.management.NotCompliantMBeanException;
+import javax.management.ObjectName;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.commons.collections.MapUtils;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.codice.ddf.persistence.attributes.AttributesStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DataUsage extends RouteBuilder implements DataUsageMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataUsage.class);
+
+    // 23:30 Daily
+    public static final String DEFAULT_CRON_TIME = "0+30+23+*+*+?";
+
+    public static final String CRON_TIME_ID_KEY = "data-usage-cron-key";
+
+    public static final String CRON_TIME_KEY = "cron_time";
+
+    public static final String ID = "id";
+
+    public static final String TXT_PREFIX = "_txt";
+
+    private static final String PREFERENCES_TYPE = "preferences";
+
+    private static final String LNG_PREFIX = "_lng";
+
+    private AttributesStore attributesStore;
+
+    private PersistentStore persistentStore;
+
+    private ObjectName objectName;
+
+    private MBeanServer mBeanServer;
+
+    private String cronTime;
+
+    private ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+
+    public DataUsage() {
+        registerMbean();
+    }
+
+    public void init() {
+        this.cronTime = getPersistentCronTime();
+        LOGGER.debug("Set Cron Time as : {}", this.cronTime);
+    }
+
+    private String getPersistentCronTime() {
+        try {
+            List<Map<String, Object>> mapList;
+            try {
+                readWriteLock.readLock()
+                        .lock();
+                mapList = persistentStore.get(PREFERENCES_TYPE);
+            } finally {
+                readWriteLock.readLock()
+                        .unlock();
+            }
+            if (mapList == null) {
+                throw new PersistenceException("Unable to get Cron Time.");
+            }
+            for (Map<String, Object> preference : mapList) {
+                String id = (String) preference.get(ID + TXT_PREFIX);
+                if (CRON_TIME_ID_KEY.equals(id)) {
+                    return (String) preference.get(CRON_TIME_KEY + TXT_PREFIX);
+                }
+            }
+        } catch (PersistenceException e) {
+            LOGGER.error("Unable to get Cron Time from Persistent Store.", e);
+        }
+        return DEFAULT_CRON_TIME;
+    }
+
+    private void setPersistentCronTime() {
+        PersistentItem persistentItem = new PersistentItem();
+        persistentItem.addIdProperty(CRON_TIME_ID_KEY);
+        persistentItem.addProperty(CRON_TIME_KEY, this.cronTime);
+
+        try {
+            readWriteLock.readLock()
+                    .lock();
```

Comment:
lock() can block indefinitely, but can that condition happen with the context of how this code runs? If yes, then one of the tryLock() variants would be a good choice.